js_modules/dagit/packages/core/src/ui/BaseButton.tsx

```diff
@@ -0,0 +1,147 @@
+import * as React from 'react';
+import styled, {css} from 'styled-components/macro';
+
+import {ColorsWIP} from './Colors';
+import {IconWrapper} from './Icon';
+
+interface Props extends React.ComponentPropsWithRef<'button'> {
+  icon?: React.ReactNode;
+  label?: React.ReactNode;
+  loading?: boolean;
+  rightIcon?: React.ReactNode;
+  fillColor?: string;
+  stroke?: boolean;
+  textColor?: string;
+}
+
+export const BaseButton = React.forwardRef(
+  (props: Props, ref: React.ForwardedRef<HTMLButtonElement>) => {
+    const {
+      fillColor = ColorsWIP.White,
+      disabled,
+      icon,
+      label,
+      loading,
+      rightIcon,
+      textColor = ColorsWIP.Dark,
```

Comment:
It's a tiny bit complicated that there are defaults here and also in the StyledButton's `textColorCSS()` helper.  I wonder if the StyledButton could require it's props if you decide to make a button from scratch